lib/ansible/plugins/action/__init__.py

```diff
@@ -611,10 +611,10 @@ def _low_level_execute_command(self, cmd, sudoable=True, in_data=None, executabl
             display.debug("_low_level_execute_command(): using become for this command")
             cmd = self._play_context.make_become_cmd(cmd, executable=executable)
 
-        if self._connection.allow_executable:
-            if executable is None:
-                executable = self._play_context.executable
-            cmd = executable + ' -c ' + pipes.quote(cmd)
+        if executable is None:
```

Comment:
this breaks `raw` and `script` which should not have a shell unless they pass it